### PR TITLE
Added "gets or sets" rule for the model template in the CSharp code generator

### PR DESCRIPTION
**Pull request by brnleehng**
Four changes:
1 - Added more correct doc strings
2 - Capitalize some abbreviations such as JSON and HTTP
3 - Created a rule for "gets or sets" in the model template, however it does not cover the case where the first word such as "Azure" or any words that should be capitalize 
4 - Regenerated expected tests

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@brnleehng__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (brhoan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by amarzavery** [comment]
@azuresdkci - Test this please!

**Comment by brnleehng** [comment]
@amarzavery Travis build passed, but the Jenkins build failed. Is there some configuration issues for Jenkins ?

**Comment by brnleehng** [comment]
Nevermind. Found the issue, will fix it

**Comment by azurecla** [comment]
Hi __@brnleehng__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (brhoan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by brnleehng** [comment]
Ran a Jenkins job (http://azuresdkci.cloudapp.net/view/3-AutoRest/job/autorest/2261/)

**Comment by amarzavery** [comment]
@azuresdkci - Add to whitelist

**Comment by jianghaolu** [comment]
@azuresdkci retest this please

**Comment by tbombach** [comment]
This PR addresses the "get or set prefix" issue from https://github.com/Azure/autorest/issues/981
(@brnleehng That way this PR shows up on the issue if someone wants to address the other points in that issue)

File: AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs [hunk 1]
```diff
@@ -137,8 +137,13 @@ private static bool ShouldValidate(this IType model)
         /// </summary>
         /// <param name="property">The given property documentation to format</param>
         /// <returns></returns>
-        public static string GetFormattedPropertyDocumentation(PropertyTemplateModel property)
+        public static string GetFormattedPropertyDocumentation(Property property)
```
Comment (tbombach) [inline AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs]: Could you make this an extension method like the other methods in this file? That way you can just call property.GetFormattedPropertyDocumentation()

File: AutoRest/Generators/CSharp/CSharp/Templates/ModelTemplate.cshtml [hunk 2]
```diff
@@ -2,6 +2,7 @@
 @using Microsoft.Rest.Generator.ClientModel
 @using Microsoft.Rest.Generator.Utilities
 @using Microsoft.Rest.Generator
+@using Microsoft.Rest.Generator.CSharp
```
Comment (tbombach) [inline AutoRest/Generators/CSharp/CSharp/Templates/ModelTemplate.cshtml]: Is this using necessary? The ShouldValidateChain() method is called in this template and is also defined in ClientModelExtensions.cs, but we didn't need the using before.

File: AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs [hunk 3]
```diff
@@ -132,6 +132,36 @@ private static bool ShouldValidate(this IType model)
...
+        public static string GetFormattedPropertyDocumentation(PropertyTemplateModel property)
+        {
+            if (string.IsNullOrEmpty(property.Documentation))
+            {
+                return property.Documentation.EscapeXmlComment();
+            }
+
+            string documentation = property.IsReadOnly ? "Gets " : "Gets or sets ";
+
+            string firstWord = property.Documentation.Split(' ').First();
```
Comment (John-Hart) [inline AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs]: Should you trim the start of the Documetation to ensure there aren't any leading spaces?

File: AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs (hunk 1 above)
Comment (brnleehng) [inline AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs]: Okay will make this change

File: AutoRest/Generators/CSharp/CSharp/Templates/ModelTemplate.cshtml (hunk 2 above)
Comment (brnleehng) [inline AutoRest/Generators/CSharp/CSharp/Templates/ModelTemplate.cshtml]: Removed

File: AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs (hunk 3 above)
Comment (brnleehng) [inline AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs]: Seems like regenerating expected tests are failing due to the trimming.

File: AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs [hunk 4]
```diff
@@ -132,6 +132,36 @@ private static bool ShouldValidate(this IType model)
         }
 
         /// <summary>
+        /// Format the documentation of a property properly with the correct getters and setters. Note that this validation will
+        /// only acryonyms and article words.
```
Comment (tbombach) [inline AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs]: Minor issue: looks like the last sentence of this method summary has a missing word and typo (acronyms).

**Comment by tbombach** [comment]
@brnleehng Sorry for not catching it in the first pass, but the only remaining thing looks like a minor typo in one of the method descriptions. Everything else looks good. Do you mind updating the PR for this? I'll merge it as soon as it passes the tests again.

**Comment by hovsepm** [comment]
same as https://github.com/Azure/autorest/pull/1035

**Pull request closed without merging**